src/app/dashboard/residence-view.component.tsx

```diff
@@ -20,16 +22,50 @@ interface MachinesListProps {
 }
 
 const MachinesList = (props: MachinesListProps) => {
+  const [showDialog, setShowDialog] = React.useState(false);
+  const [machine, setMachine] = React.useState<Machine>();
+
+  const handleOpenDialog = (machine: Machine) => () => {
+    setShowDialog(true);
+    setMachine(machine);
+  };
+
+  const handleSetTime = (minutes: number) =>  {
+    if (machine !== undefined) {
+      machine.deadline = getDelayedDateByMinutes(new Date(), minutes); //TODO: set to database
+    }
+    setShowDialog(false);
+    startTick();
+  };
+
+  const handleCancelClick = () => {
+    setShowDialog(false);
+  };
+
+  const decrementDeadline = (deadline: Date) => {
+    deadline = tick(deadline);
+  };
+
+  const startTick = () => {
+    if (machine !== undefined) {
+      const callback = () => decrementDeadline(machine.deadline);
+      setInterval(callback, 1000);
```

Comment:
Para evitar memory leak é preciso salvar a função do setInternal em uma variável e limpa-la quando o componente for desmontado